src/dmd/cppmanglewin.d

```diff
@@ -388,11 +390,12 @@ public:
             return;
         //printf("visit(TypeStruct); is_not_top_type = %d\n", (int)(flags & IS_NOT_TOP_TYPE));
         mangleModifier(type);
-        if (type.sym.isUnionDeclaration())
+        auto agg = cast(StructDeclaration) type.sym;
+        if (agg.isUnionDeclaration())
             buf.writeByte('T');
         else
-            buf.writeByte(type.cppmangle == CPPMANGLE.asClass ? 'V' : 'U');
-        mangleIdent(type.sym);
+            buf.writeByte(agg.cppmangle == CPPMANGLE.asClass ? 'V' : 'U');
```

Comment:
`.cppmangle` is a field in `AggregateDeclaration`, but the only use of `agg` is to get that field. Shouldn't `agg` be an `AggregateDeclaration`, not a `StructDeclaration` ?

Reply:
It could be, but since we are in `visit(TypeStruct)`, we know that it's a `StructDeclaration` already, and I thought it'd be less confusing.